src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1864,7 +1879,7 @@ public static string Mode(PSObject instance)
             // Mark the last bit as a "l" if it's a reparsepoint (symbolic link or junction)
             // Porting note: these need to be handled specially
             bool isReparsePoint = InternalSymbolicLinkLinkCodeMethods.IsReparsePoint(fileInfo);
-            bool isHardLink = InternalSymbolicLinkLinkCodeMethods.IsHardLink(fileInfo);
+            bool isHardLink = includeHardLink ? InternalSymbolicLinkLinkCodeMethods.IsHardLink(fileInfo) : false;
             mode[5] = isReparsePoint || isHardLink ? 'l' : '-';
```

Comment:
Might be better to not include the `-` if `ModeWithoutHardlink` is used to not imply it's not a hardlink.

Reply:
True. Will fix

Reply:
@SteveL-MSFT  The result may then look like this:

```output
Mode                LastWriteTime         Length Name
----                -------------         ------ ----
d----         2019-01-29    15:29                .vsts-ci
d----l        2019-01-30    17:44                a
d----         2018-06-18    23:11                demos
```
`a` is a symlink, but since we don't show the '-' it will be hanging.

Reply:
Let's have @PowerShell/powershell-committee review this and propose something

Reply:
@SteveL-MSFT And when it is a file? That is hardlinked?

This is how they look today.
```output
-a---l        2018-09-15    18:29         316640 WMSysPr9.prx
-a---l        2018-09-15    09:29          11264 write.exe
```
Should that be like this?
```output
la---        2018-09-15    18:29         316640 WMSysPr9.prx
la---        2018-09-15    09:29          11264 write.exe
```

Reply:
@powercode yes (`la---`), the intent is to reuse the 'directory' column since a link is a different type that a file or directory.

Reply:
I believe this is resolved.  Following the Unix `ls -l` model, a link is distinct from a file or directory.  You don't know what it points to until you resolve it.  `ls -l` solves this with the `link -> target` format which you've added.